src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -126,9 +160,11 @@ void rotate(@Nullable Long offset) throws IOException, DataException {
     void finishFile(Boolean delete) throws IOException, DataException {
         if (isDirty()) {
             recordWriter.commit();
-            GZIPOutputStream gzip = (GZIPOutputStream) outputStream;
-            gzip.finish();
-
+            // Since we are using GZIP compression, finish the file. Close is invoked only when this flush finishes
+            // and then the file is finished in ingest
+            // This is called when there is a time or a size limit reached. The file is then reset/rolled and then a
+            // new file is started for processing

```

Comment:
```suggestion
            // new file is created for processing

```

Reply:
Good point Ohad

Path path = Paths.get(tempDirPath, tempDir).toAbsolutePath(); made this fix in TopicPartitionWriter